### PR TITLE
Add npm Build/Watch Scripts in `utils/build` for Node users

### PR DESCRIPTION
**Pull request by mattdesl**
This improves the "Quickstart (Node.js)" process for those developing ThreeJS with tools like Node/npm/browserify/webpack/etc.

https://github.com/mrdoob/three.js/wiki/build.py,-or-how-to-generate-a-compressed-Three.js-file#quickstart-nodejs

Changes:
- Adds a `npm run build` command that handles the commons/defaults.
- Adds a `npm run watch` for development, so that changing files in `src/**` will re-run the above `build` script
- Reformatted the package.json to use two spaces, since this is the default with tooling like `npm install --save`

This makes it a bit easier to develop ThreeJS code within an application. You can symlink and/or `require()` the build file and changing ThreeJS source will trigger a re-build.

---

To recap, the build/dev commands are now:

```
# initial setup
npm install

# build JS bundle
npm run build

# or, you can still call the script directly
node build.js --include common --include extras

# for development, rebuild on 'src/**' file changes
npm run watch
```

### CODE REVIEW

**Comment by bhouston** [comment]
I personally would like to move away from python as the build language and just support node.js-based build script.  Less stuff to maintain.

**Comment by mrdoob** [comment]
Was the tabs to spaces change intentional?

**Comment by mrdoob** [comment]
> I personally would like to move away from python as the build language and just support node.js-based build script. Less stuff to maintain.

How are the node.js closure compiler alternatives doing?

**Comment by bhouston** [comment]
I would still use the closure compiler btw, just call it from NodeJS rather than Python.  Less code to maintain.  Someone did a recent test and Closure Compiler came out on top:

https://github.com/samccone/The-cost-of-transpiling-es2015-in-2016

Further off topic: the latest Google closure compiler releases from December 2015 now support ES6+ classes and other things and it cross compiles down to the ES5.  This means that one could start to use ES6+ constructs in ThreeJS and still be compatible with older browsers.  I'm not saying we should spent a lot of time on this, but it may be something to start considering.

**Comment by mrdoob** [comment]
Ah, I see. Yeah, happy to move to node.js. Right now build.js uses uglify though. Any chance someone could update it so it uses closure compiler?

**Comment by bhouston** [comment]
> Any chance someone could update it so it uses closure compiler?

This is a really good idea.  We could make the packaging a configurable option as well.

As we refactor this we might as well use the npm closure-compiler module rather than including it in the threeJS repo - make things smaller and more consistent:

https://www.npmjs.com/package/closure-compiler (there are a few npm modules for the closure compiler, but this is the most popular.)

**Comment by Mugen87** [comment]
see #10860
